src/app/zap-templates/call-command-handler-src.zapt

```diff
@@ -100,15 +100,49 @@ EmberAfStatus emberAf{{asCamelCased name false}}Cluster{{asCamelCased side false
         {{#if (isCommandAvailable parent.side incoming outgoing commandSource name)}}
         case ZCL_{{asDelimitedMacro name}}_COMMAND_ID: {
             {{#if (zcl_command_arguments_count this.id)}}
-            uint32_t argOffset = 0;
+            uint32_t payloadOffset = cmd->payloadStartIndex;
             {{#zcl_command_arguments}}
-            {{asUnderlyingType type}} * {{asSymbol label}} = ({{asUnderlyingType type}} *)(cmd->buffer + argOffset);
+            {{#if (isNullable type isArray)}}
+            uint8_t * {{asSymbol label}};
+            {{else}}
+            {{asUnderlyingType type}} {{asSymbol label}};
+            {{/if}}
+            {{/zcl_command_arguments}}
+            {{#zcl_command_arguments}}
+            {{#if presentIf}}
+            if ({{presentIf}})
```

Comment:
The whole presentIf part seems a little bit shaky to me. First of all, all of the conditional element are in text in the XML files and transposing them directly into code might not work at all. Secondly while this condition is actually written in C inside the Silabs XML, it is not the case in the official DotDot XML. So switching between the two will be harder in the future.

Reply:
I think that's good enough for the time being. But in the long term, yes something more elaborate needs to be supported.